user_profile_modal: Fix inconsistent word breaks in profile fields.

sayamsamal opened the pull request:
We used to use `word-break: breal-all` for the profile fields in the user profile modal. This property used to insert word break in between any two characters to prevent overflow, which lead to issues with the label names in non-English languages and overall reduced experience while reading other long text profile fields.

`overflow-wrap: anywhere` tries to break a long word or URL at an acceptable break point, if there exists one. In case there is no acceptable breakpoints only then is the word break inserted between any two characters.

Fixes #22865.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

alya commented:
Thanks!

I'm trying to understand the PR description, and I don't follow why the fields fit better now. Does this also give more space to the labels?

sayamsamal commented:
@alya Yes, Since I have moved from word wrap to overflow-wrap, I have also removed the fixed width constraint of the labels and profile fields, they are now able to just fill up the remaining space as shown in the screenshot below and in the PR description.

![image](https://user-images.githubusercontent.com/82862779/202557891-9ba0c699-69fb-4df9-96be-b5f4c87b77fa.png)

alya commented:
I would leave a bit more margin between the text and the avatar; otherwise looks good to me!

@sahil839 Would you be up for reviewing this one?

sahil839 commented on static/styles/popovers.css:
```diff
@@ -450,7 +449,7 @@ ul {
     .col-left {
         padding: 0 10px 0 0;
-        word-break: break-all;
+        overflow-wrap: anywhere;
```
Would probably be better to use `brea-word` value for `overflow-wrap` instead of `anywhere`. The [docs](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap#browser_compatibility) mention that `anywhere` is not supported by Safari browser. And the behavior should be same for both values since we are not setting width to `min-content` in this case.

timabbott commented:
@sayamsamal do you have a chance to reply to the feedback here?

sahil839 commented:
@sayamsamal do you have time to update this?

timabbott commented:
Closing as the issue was completed through another PR.
